itou/files/management/commands/delete_unused_files.py

```diff
@@ -0,0 +1,83 @@
+import datetime
+import functools
+import operator
+
+from django.conf import settings
+from django.utils import timezone
+
+from itou.antivirus.models import Scan
+from itou.files.models import File
+from itou.utils.command import BaseCommand
+from itou.utils.storage.s3 import TEMPORARY_STORAGE_PREFIX, s3_client
+
+
+S3_CLEANING_DELAY = datetime.timedelta(days=1)
+
+
+class Command(BaseCommand):
+    def get_relations(self):
+        relations = {
+            (remote_field.field.model, remote_field.field.name)
+            for remote_field in File._meta.get_fields(include_hidden=True)
+            if remote_field.is_relation
+        }
+        relations.remove((Scan, "file"))
+        return relations
+
+    def delete_orphan_files(self):
+        linked_files_pks = functools.reduce(
+            operator.or_,
+            [
+                set(model.objects.exclude(**{field: None}).values_list(field, flat=True))
+                for model, field in self.get_relations()
+            ],
+        )
+        deletions, deletions_per_type = (
+            File.objects.filter(last_modified__lte=timezone.now() - datetime.timedelta(days=1))
+            .exclude(pk__in=linked_files_pks)
+            .delete()
+        )
+        if deletions:
+            self.logger.info(f"Deleted {deletions_per_type['files.File']} orphans files from database")
+        else:
+            self.logger.info("No orphan file found in database")
```

Comment:
Au pire ça mettra 0, non ?
```suggestion
        self.logger.info(f"Deleted {deletions_per_type['files.File']} orphans files from database")
```